### PR TITLE
Save error code in the event when rejecting envelopes

### PR DESCRIPTION
**Pull request by aliveni**
Reopening #452 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/NotificationServiceTest.java [hunk 1]
```diff
@@ -104,7 +105,7 @@ void should_not_send_notification_when_notification_was_already_sent() {
         // given
         // rejected and notification_sent
         var envelopeId1 = envelopeService.createNewEnvelope("bulkscan", "blob1.zip", now());
-        envelopeService.markAsRejected(envelopeId1, "duplicate file");
+        envelopeService.markAsRejected(envelopeId1, ErrorCode.ERR_METAFILE_INVALID, "duplicate file");
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/NotificationServiceTest.java:108]: blob router does not know about metafile, a different code should be used

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/BlobVerifier.java [hunk 2]
```diff
@@ -44,31 +45,33 @@ public VerificationResult verifyZip(String blobName, byte[] rawBlob) {
             return ok();
         } catch (DocSignatureFailureException ex) {
             logger.info("Invalid signature. Blob name: {}", blobName, ex);
-            return error("Invalid signature");
+            return error(ErrorCode.ERR_SIG_VERIFY_FAILED, "Invalid signature");
         } catch (InvalidZipArchiveException ex) {
             logger.info("Invalid zip archive. Blob name: {}", blobName, ex);
-            return error("Invalid zip archive");
+            return error(ErrorCode.ERR_METAFILE_INVALID, "Invalid zip archive");
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/BlobVerifier.java:51]: maybe `ERR_ZIP_PROCESSING_FAILED`?

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/NotificationServiceTest.java (hunk 1 above)
Comment (aliveni) [reply src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/NotificationServiceTest.java:108]: update to use `ERR_ZIP_PROCESSING_FAILED` error code

**Comment by aliveni** [comment]
PR build is failing. I'll re-open a new PR

**Pull request closed without merging**